### PR TITLE
Fixes #14891: Rtf can not launch ncf tests on non-released major rudder version

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/14891

### CODE REVIEW

File: scripts/rudder-setup/add-repo.sh [hunk 1]
```diff
@@ -25,11 +25,7 @@ add_repo() {
 
   if [ "${USE_CI}" = "yes" ]
   then
-    if [ -z "$(echo "${RUDDER_VERSION}" | sed -e '/\..*\./d')" ]; then
-      $local URL_BASE="http${S}://publisher.normation.com/${REPO_PREFIX}${REPO_TYPE}/${RUDDER_VERSION}-nightly"
-    else
       $local URL_BASE="http${S}://publisher.normation.com/${REPO_PREFIX}${REPO_TYPE}/${RUDDER_VERSION}"
```
Comment (peckpeck) [inline scripts/rudder-setup/add-repo.sh]: indent

File: scripts/rudder-setup/rudder-setup-src.sh [hunk 2]
```diff
@@ -10,14 +10,14 @@ usage() {
...
-  echo "       x.y-nightly: the latest public x.y nightly build (ex: 3.2-nightly)"
-  echo "       ci/x.y:      the latest development x.y nightly build (ex: ci/3.2)"
-  echo "       ci/x.y.z:    the latest development x.y.z release build (ex: ci/3.2.16)"
-  echo "       ci/x.y.z~a:  the latest development x.y.z pre-release build (ex: ci/4.0.0~rc1)"
-  echo "       latest:      the latest stable version"
+  echo "       x.y:            the latest x.y release (ex: 3.2)"
+  echo "       x.y.z:          the exact x.y.z release (ex: 3.2.1)"
+  echo "       x.y.z~a:        the latest x.y.z pre-release where a can be alpha1, beta1, rc1... (ex: 4.0.0~rc1) "
+  echo "       x.y-nightly:    the latest public x.y nightly build (ex: 3.2-nightly)"
+  echo "       ci/x.y-nightly: the latest development x.y nightly build (ex: ci/3.2)"
```
Comment (peckpeck) [inline scripts/rudder-setup/rudder-setup-src.sh]: use the same order as above

File: scripts/rudder-setup/setup-ncf.sh [hunk 3]
```diff
@@ -45,7 +45,7 @@ setup_ncf() {
   if echo "${CFENGINE_VERSION}" | grep "rudder" >/dev/null
   then
     # cfengine from Rudder
-    RUDDER_VERSION=$(echo "${CFENGINE_VERSION}" | cut -f 2 -d "-")
+    RUDDER_VERSION=$(echo "${CFENGINE_VERSION}" | cut -f2- -d "-")
```
Comment (peckpeck) [inline scripts/rudder-setup/setup-ncf.sh]: wut ?

**Comment by Fdall** [comment]
Commit modified

File: scripts/rudder-setup/rudder-setup-src.sh [hunk 4]
```diff
@@ -10,14 +10,14 @@ usage() {
...
-  echo "       ci/x.y.z:    the latest development x.y.z release build (ex: ci/3.2.16)"
-  echo "       ci/x.y.z~a:  the latest development x.y.z pre-release build (ex: ci/4.0.0~rc1)"
-  echo "       latest:      the latest stable version"
+  echo "       x.y:            the latest x.y release (ex: 3.2)"
+  echo "       x.y.z:          the exact x.y.z release (ex: 3.2.1)"
+  echo "       x.y.z~a:        the latest x.y.z pre-release where a can be alpha1, beta1, rc1... (ex: 4.0.0~rc1) "
+  echo "       x.y-nightly:    the latest public x.y nightly build (ex: 3.2-nightly)"
+  echo "       ci/x.y.z:       the latest development x.y.z release build (ex: ci/3.2.16)"
+  echo "       ci/x.y.z~a:     the latest development x.y.z pre-release build (ex: ci/4.0.0~rc1)"
+  echo "       ci/x.y-nightly: the latest development x.y nightly build (ex: ci/3.2)"
```
Comment (peckpeck) [inline scripts/rudder-setup/rudder-setup-src.sh]: the example doesn't match

**Comment by Fdall** [comment]
Commit modified

**Pull request merged**